File: app/src/main/java/io/datalive/myreferrer/FirstInstallActivity.java
package io.datalive.myreferrer;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class FirstInstallActivity extends AppCompatActivity {

    TextView txtReferrer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_first_install);

        txtReferrer = (TextView) findViewById(R.id.txtReferrer);

        Intent intent = getIntent();
        try {
            String referrer = intent.getStringExtra("referrer");

            txtReferrer.setText("Referrer : " + referrer);
        } catch (NullPointerException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/io/datalive/myreferrer/FirstInstallActivity.kt
package io.datalive.myreferrer

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_first_install.*

class FirstInstallActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_first_install)

        intent.getStringExtra("referrer").let {
            txtReferrer.text = "Referrer : $it"
        }
    }
}
